src/embeddings/embeddings_api.cpp

```diff
@@ -0,0 +1,69 @@
+//*****************************************************************************
+// Copyright 2024 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "embeddings_api.hpp"
+
+#include <string>
+#include <variant>
+
+#include "rapidjson/document.h"
+
+std::variant<EmbeddingsRequest, std::string> EmbeddingsRequest::from_json(rapidjson::Document* parsedJson) {
+    EmbeddingsRequest request;
+    std::vector<std::string> input_strings;
+
+    if (!parsedJson->IsObject())
+        return "Received json is not an object";
```

Comment:
This is very nice use of std::variant, however I think we should adhere to common patterns in this project which is returning status - either ovms::Status or abseil status in case it is more convenient for calculator
@atobiszei @bstrzele what do you think?